File: 0066-plus-one/0066-plus-one.java
class Solution {
    public int[] plusOne(int[] d) {
        int[] res = new int[d.length + 1];
        int lst=d[d.length-1]+1;
        int carry = 0;
        if(lst<=9){
            d[d.length-1]=lst;
            return d;
        }
        else{
            carry=1;
        for (int i = d.length-1; i >= 0; i--) {
            if((carry+d[i])==10){
                res[i+1]=(carry+d[i])%10;
                carry=(carry+d[i])/10;
            }
            else{
                res[i+1]=(carry+d[i]);
                carry=(carry+d[i])/10;
            }
        }
    }
        if(carry>0){
            res[0]=1;
            return res;
        }
        else{
             int[] trim=Arrays.copyOfRange(res,1,res.length);
             return trim;
        }
       
    }
}

File: 0066-plus-one/0066-plus-one.py
class Solution(object):
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        carry=1
        l=[]
        for i in range(len(digits)-1,-1,-1):
            v=digits[i]+carry
            digits[i]=v%10
            carry=v//10
        if carry>0:
            digits.insert(0,carry)
        return digits
            